### PR TITLE
SCC-4227 - Bib Page Electronic Resources

### PR DESCRIPTION
**Pull request by dgcohen**
## Ticket:

- JIRA ticket [SCC-4227](https://newyorkpubliclibrary.atlassian.net/browse/SCC-4227)

## This PR does the following:

- Adds a new component ElectronicResources and relevant unit tests
- Adds logic to Bib page to check hasElectronicResources before rendering ElectronicResources

## How has this been tested?

- Added unit tests
- Updated Bib page integration tests

## Accessibility concerns or updates

- Should be reviewed by Clare

### Checklist:

- [x] I updated the CHANGELOG with the appropriate information and JIRA ticket number (if applicable).
- [x] I have added relevant accessibility documentation for this pull request.
- [x] All new and existing tests passed.

[SCC-4227]: https://newyorkpubliclibrary.atlassian.net/browse/SCC-4227?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by charmingduchess — APPROVED** [review]

**Review by EdwinGuzman — COMMENTED** [review]
The conditional logic works but the component itself is not correct. I think it's within scope to fix it. The issue is that the component is reused in the Search Results page and in the Bib page but perhaps it shouldn't be. 

I think a _new_ component is needed that is similar or the content should be updated. The links in the Bib page should point to the actual online resource but it's still saying "view all available online resources >".

A few examples:
- Prod: https://www.nypl.org/research/research-catalog/bib/b13493199
- PR: https://research-catalog-git-scc-4227-fix-eresources-link-logic-nypl.vercel.app/research/research-catalog/bib/b13493199

- Prod: https://www.nypl.org/research/research-catalog/bib/b21406328
- PR: https://research-catalog-git-scc-4227-fix-eresources-link-logic-nypl.vercel.app/research/research-catalog/bib/b21406328

- Prod: https://www.nypl.org/research/research-catalog/bib/b21244022
- PR: https://research-catalog-git-scc-4227-fix-eresources-link-logic-nypl.vercel.app/research/research-catalog/bib/b21244022#electronic-resources


(Also, the title in RC is not correct compared to prod but that seems like a minor and separate issue but just want to flag it now.)

**Review by EdwinGuzman — CHANGES_REQUESTED** [review]
Forgot to request changes.

**Comment by dgcohen** [comment]
> The conditional logic works but the component itself is not correct. I think it's within scope to fix it. The issue is that the component is reused in the Search Results page and in the Bib page but perhaps it shouldn't be.
> 
> I think a _new_ component is needed that is similar or the content should be updated. The links in the Bib page should point to the actual online resource but it's still saying "view all available online resources >".
> 
> A few examples:
> 
> * Prod: https://www.nypl.org/research/research-catalog/bib/b13493199
> * PR: https://research-catalog-git-scc-4227-fix-eresources-link-logic-nypl.vercel.app/research/research-catalog/bib/b13493199
> * Prod: https://www.nypl.org/research/research-catalog/bib/b21406328
> * PR: https://research-catalog-git-scc-4227-fix-eresources-link-logic-nypl.vercel.app/research/research-catalog/bib/b21406328
> * Prod: https://www.nypl.org/research/research-catalog/bib/b21244022
> * PR: https://research-catalog-git-scc-4227-fix-eresources-link-logic-nypl.vercel.app/research/research-catalog/bib/b21244022#electronic-resources
> 
> (Also, the title in RC is not correct compared to prod but that seems like a minor and separate issue but just want to flag it now.)

You're right that this is a separate component, can do it as part of this ticket since it's not very big. Just update the title and description of the ticket.

File: CHANGELOG [hunk 1]
```diff
@@ -18,6 +24,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Fixed
 
+- Added check for hasElectronicResources in Bib Page before rendering ElectronicResources.
```
Comment (EdwinGuzman) [inline CHANGELOG:27]: Move this under prerelease.

**Review by EdwinGuzman — APPROVED** [review]
It looks good, but note that the pattern in  `electronicResourcesList` is what we were trying to move away from: keeping data logic and rendering specific HTML together. I don't think it's a big issue right now and it would also be fine if it always renders a list, even if there is only one electronic resource.

**Comment by dgcohen** [comment]
> It looks good, but note that the pattern in `electronicResourcesList` is what we were trying to move away from: keeping data logic and rendering specific HTML together. I don't think it's a big issue right now and it would also be fine if it always renders a list, even if there is only one electronic resource.

Good call, I removed electronicResourcesList and have it always rendering a list. Much cleaner now overall.

**Pull request merged**